src/web/js/ide.js

```diff
@@ -79,7 +106,28 @@ const runtimeApi = {
              (TODO: return richer values for error returns)
   */
   execute(bytecode) {
-    return new Promise((resolve, reject) => reject(new Error("Not Implemented")));
+    var runtime = window.CPOIDEHooks.runtime;
```

Comment:
rather than proliferating references to global `window` properties, I think it would be be preferable to a narrow "window" where we hook up to this global. e.g. something like this:

```javascript
function createRuntimeApi(CPOIDEHooks) {
  return {
    parse() { ... },
    compile() { ... },
    execute() {
      var runtime = CPOIDEHooks.runtime;
      ...
    },
  };
}
```
And then on line 150, you'd have:
```javascript
           resolve(createRuntimeApi(window.CPOIDEHooks));
```